### ISSUE TITLE
False positive for shaders of different stages linked into a common module and/or using a common set of descriptors

### ISSUE THREAD

**Issue by hawk1x** [KhronosGroup/Vulkan-ValidationLayers#10035]
**Environment:**
 - OS: windows 10
 - GPU and driver version: nvidia gtx950 576.02
 - SDK or header version if building from repo: 1.4.309.0
 - Options enabled (synchronization, best practices, etc.): synchronization, best practices arm

**Describe the Issue**

The compute shader does not use images, it is simply linked with the fragment shader into one module and uses a common set of descriptors.

**Expected behavior**

--

**Valid Usage ID**

```sh
Validation Performance Warning: [ BestPractices-Arm-vkCreateComputePipelines-compute-spatial-locality ] | MessageID = 0x6d8fa932
vkCreateComputePipelines(): pCreateInfos[0] [Arm] compute shader has work group dimensions (52, 1, 1), which suggests a 1D dispatch, but the shader is accessing 2D or 3D images. The shader may be exhibiting poor spatial locality with respect to one or more shader resources.
```

**Additional context**

--

**Comment by spencer-lunarg** [comment]
@hawk1x I have a fix here https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/10040

could you share the SPIR-V that was causing the issue, can plug it in and confirm it is also not going to report a false positive

**Comment by hawk1x** [comment]
Sure

[out.zip](https://github.com/user-attachments/files/19961312/out.zip)
